zk-token-sdk/src/errors.rs

```diff
@@ -6,6 +6,8 @@ use {
 
 #[derive(Error, Clone, Debug, Eq, PartialEq)]
 pub enum ProofError {
+    #[error("invalid transfer amount range")]
+    TransferAmount,
```

Comment:
fyi: this is a little dangerous.  It's ok for now as the zk proof program isn't yet activated, but if it was then this would break consensus. 


Instead new enum entries are always added at the end.  If the zk proof program was already active then we'd need a new feature to activate the use of this new error code